File: Stocks/Stocks.Model/Stock.cs
namespace Stocks.Model
{
    public class Stock
    {
        public Guid Id { get; set; }
        public string Symbol { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public double CurrentPrice { get; set; }
        public long MarketCap { get; set; }
        public Guid TraderId { get; set; }

    }
}

File: frontend/react-app/stocks-react/src/Stock.js
import React from 'react'
import { Link } from 'react-router-dom'

export default function Stock({symbol, companyName, currentPrice, marketCap, index, deleteItem, user}) {
  return (
    <tr>
      <td>{symbol}</td>
      <td>{companyName}</td>
      <td>{currentPrice}</td>
      <td>{marketCap}</td>
      {
           user.role === 'admin' ? 
           <td>
            <Link to={`/Edit/${index}`}>Edit</Link>
            <button onClick={() => deleteItem(index)}>Delete</button>
           </td>:
           null
        }
    </tr>
  )
}

